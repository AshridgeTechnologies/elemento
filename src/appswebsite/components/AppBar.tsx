import * as React from 'react';
import Mui_AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import UserMenu from '../../shared/UserMenu'

export default function AppBar() {
    return (
        <Box sx={{ flexGrow: 1 }}>
            <Mui_AppBar position="sticky">
                <Toolbar variant="dense">
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{ mr: 2 }}
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" component="div" sx={{ fontSize: 16, flexGrow: 1 }}>
                        Elemento Apps
                    </Typography>
                    <UserMenu/>
                </Toolbar>
            </Mui_AppBar>
        </Box>
    );
}