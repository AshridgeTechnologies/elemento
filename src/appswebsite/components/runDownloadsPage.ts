import thePage from './DownloadsPage'
import {run} from '../../appsShared/renderInPage'

run(thePage)
