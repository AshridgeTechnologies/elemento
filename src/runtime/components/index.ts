export {default as App} from './App'
export {default as Button} from './Button'
export {default as Icon} from './Icon'
export {default as Image} from './Image'
export {default as UserLogon} from './UserLogon'
export {default as Menu} from './Menu'
export {default as MenuItem} from './MenuItem'
export {default as NumberInput} from './NumberInput'
export {default as DateInput} from './DateInput'
export {default as TrueFalseInput} from './TrueFalseInput'
export {default as SpeechInput} from './SpeechInput'
export {default as Data} from './Data'
export {default as Calculation} from './Calculation'
export {default as Timer} from './Timer'
export {default as Collection} from './Collection'
export {default as Page} from './Page'
export {default as ComponentElement} from './ComponentElement'
export {default as Form} from './Form'
export {default as BaseFormState} from './FormState'
export {default as SelectInput} from './SelectInput'
export {default as TextElement} from './TextElement'
export {default as Layout} from './Layout'
export {default as Block} from './Block'
export {default as AppBar} from './AppBar'
export {default as ListElement} from './ListElement'
export {default as ListItem} from './ListItem'
export {default as ItemSet} from './ItemSet'
export {default as ItemSetItem} from './ItemSetItem'
export {default as TextInput} from './TextInput'
export {default as MemoryDataStore} from './MemoryDataStore'
export {default as FileDataStore} from './FileDataStore'
export {default as FileDataStoreImpl} from './FileDataStoreImpl'
export {default as IdbDataStoreImpl} from './IdbDataStoreImpl'
export {default as BrowserDataStore} from './BrowserDataStore'
export {default as FirestoreDataStore} from './FirestoreDataStore'
export {default as FirestoreDataStoreImpl} from './FirestoreDataStoreImpl'
export {default as ServerAppConnector} from './ServerAppConnector'
export {default as Adapter} from './Adapter'
