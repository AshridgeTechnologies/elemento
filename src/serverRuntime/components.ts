export {default as FirestoreDataStore} from './FirestoreDataStore'
export {default as Collection} from './Collection'

// NOTE: Must add the name of each component to ./names.ts