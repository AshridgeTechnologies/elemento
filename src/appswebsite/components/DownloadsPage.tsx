import React from 'react'
import {Box, Container, Link, Stack, Typography} from '@mui/material'
import AppBar from '../../shared/AppBar'
import {theme} from '../../shared/styling'
import {ThemeProvider} from '@mui/material/styles'

export default function LearnPage() {

    return <ThemeProvider theme={theme}>
        <Box display='flex' flexDirection='column' height='100%' width='100%'>
            <Box flex='0'>
                <AppBar title='Elemento Online - Downloads' userMenu={false}/>
            </Box>
            <Box flex='1' minHeight={0}>
                <Container>
                    <Stack maxWidth={600} spacing={3} mt={3}>
                        <Typography variant='h4'>Elemento Development Server</Typography>
                        <Typography>Please download the correct version for your operating system.</Typography>
                        <Typography>Your computer will complain that the app is not known, approved, signed etc.<br/>
                            We do not currently have the resources to jump through all the hoops required to get the app
                            approved on each operating system,
                            so you will just have to trust us here, and continue past all the warnings.
                        </Typography>
                        <Link href='/devServer/win/ElementoDevServer.exe'><Typography variant='h5'>Windows</Typography></Link>
                        <Link href='/devServer/macos/ElementoDevServer.zip'><Typography variant='h5'>Mac</Typography></Link>
                        <Link href='/devServer/linux/ElementoDevServer'><Typography variant='h5'>Linux</Typography></Link>
                    </Stack>
                </Container>
            </Box>
        </Box>
    </ThemeProvider>
}
