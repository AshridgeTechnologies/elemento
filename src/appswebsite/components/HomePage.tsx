import React from 'react'
import {Box, Container, Link, Stack, Typography} from '@mui/material'
import AppBar from '../../shared/AppBar'
import {theme} from '../../shared/styling'
import {ThemeProvider} from '@mui/material/styles'
import HomeIcon from '@mui/icons-material/Home'
import LightbulbIcon from '@mui/icons-material/LightbulbTwoTone'
import CreateIcon from '@mui/icons-material/CreateTwoTone'
import AodIcon from '@mui/icons-material/AodTwoTone'
import ArrowCircleRightTwoToneIcon from '@mui/icons-material/ArrowCircleRightTwoTone';

const primary = theme.palette.primary.main
const secondary = theme.palette.secondary.main

function Logo() {
    return <Box color={secondary} border='8px solid' borderRadius={4} width={108} height={104} paddingTop='10px' paddingX='3px'>
        <Stack>
            <Typography fontSize='100px' fontFamily='"Helvetica"' align='center' lineHeight={0.8}>El</Typography>
            <Typography fontSize='16px' lineHeight={1}>42</Typography>
        </Stack>
    </Box>
}

function SectionLink(props: {title: string, href: string, icon: any, children: any}) {
    const Icon = props.icon
    return <Box border='4px solid' borderColor={primary} borderRadius={4} width='300px' paddingY='10px' paddingX='16px'>
        <Link href={props.href} underline='none'>
            <Stack spacing={2}>
                <Box textAlign='center'><Icon sx={{ fontSize: 180 }} /></Box>
                <Link component='div' underline='hover' href={props.href} variant='h5' textAlign='center'>{props.title}</Link>
                <Typography fontSize='20px' color='#555'>{props.children}</Typography>
                <Box textAlign='center'><ArrowCircleRightTwoToneIcon sx={{ fontSize: 32, color: secondary }} /></Box>
            </Stack>
        </Link>
    </Box>
}

export default function HomePage() {

    return <ThemeProvider theme={theme}>
        <Box display='flex' flexDirection='column' height='100%' width='100%'>
            <Box flex='0'>
                <AppBar title='Elemento Apps - Home'/>
            </Box>
            <Box flex='1' minHeight={0} mt={2}>
                <Container maxWidth="lg">
                    <Stack spacing={2}>
                        <Stack direction='row'>
                            <Logo/>
                            <Stack ml={4}>
                                <Typography fontWeight='bold' variant='h3'  color={secondary} align='center'>Elemento</Typography>
                                <Typography variant='h4' color={secondary} align='center' mt={2}>
                                    Everything you need to make and use your own software
                                </Typography>
                            </Stack>
                        </Stack>
                        <Stack direction='row' justifyContent='space-between' alignItems='stretch' marginTop='30px'>
                            <SectionLink title='Learn' href='/learn' icon={LightbulbIcon}>
                                A great range of resources to teach you how to get started and create your first app with Elemento.
                            </SectionLink>
                            <SectionLink title='Create' href='/studio' icon={CreateIcon}>
                                Put your app together and preview it live in Elemento Studio
                            </SectionLink>
                            <SectionLink title='Use' href='/run' icon={AodIcon}>
                                Run your app - or go back to the apps you have already made.
                            </SectionLink>
                        </Stack>
                    </Stack>
                </Container>
            </Box>
        </Box>
    </ThemeProvider>
}
