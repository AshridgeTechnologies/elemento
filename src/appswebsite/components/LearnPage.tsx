import React from 'react'
import Toolbar from '@mui/material/Toolbar'
import {Box, Link, Stack, Typography} from '@mui/material'
import AppBar from '../../shared/AppBar'
import {theme} from '../../shared/styling'
import {ThemeProvider} from '@mui/material/styles'
import {TutorialRunner} from '../../tutorials/TutorialRunner'

export default function LearnPage() {

    return <ThemeProvider theme={theme}>
        <Box display='flex' flexDirection='column' height='100%' width='100%'>
            <Box flex='0'>
                <AppBar title='Elemento Apps - Learn' userMenu={false}/>
            </Box>
            <Box flex='1' minHeight={0}>
                <TutorialRunner/>
            </Box>
        </Box>
    </ThemeProvider>
}
