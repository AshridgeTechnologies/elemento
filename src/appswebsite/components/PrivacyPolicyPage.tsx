import React from 'react'
import {Box, Stack, Typography} from '@mui/material'
import AppBar from './AppBar'

export default function PrivacyPolicyPage() {

    return <Box display='flex' flexDirection='column' height='100%' width='100%'>
        <Box flex='0'>
            <AppBar/>
        </Box>
        <Box flex='1' minHeight={0}>
            <Stack spacing={2}>
                <Typography variant='h4'>Elemento Apps - Privacy Policy</Typography>
                <Typography variant='body1'>Coming soon</Typography>
            </Stack>
        </Box>
    </Box>
}
