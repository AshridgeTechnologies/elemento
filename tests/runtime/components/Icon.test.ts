/**
 * @jest-environment jsdom
 */

import {createElement} from 'react'
import {Icon} from '../../../src/runtime/components/index'
import {componentJSON, snapshot, valueObj} from '../../testutil/testHelpers'
import userEvent from '@testing-library/user-event'
import {testContainer, wait} from '../../testutil/rtlHelpers'

const doIt = () => {}

test('Icon element produces button output with properties supplied',
    snapshot(createElement(Icon, {path: 'app.page1.save', iconName: 'star', label: 'Star it', action: () => {doIt()}}))
)

test('Icon element produces empty output with display false',
    snapshot(createElement(Icon, {path: 'app.page1.save', iconName: 'star', action: () => {doIt()}, display: false}))
)

test('Icon element produces icon output with properties supplied',
    snapshot(createElement(Icon, {path: 'app.page1.save', iconName: 'star', label: 'Star it'}))
)

test('Icon element produces empty  output with display false',
    snapshot(createElement(Icon, {path: 'app.page1.save', iconName: 'star', action: () => {doIt()}, display: false}))
)

test('Icon element produces output with properties supplied as state values', async () => {
        const element = createElement(Icon, {path: 'app.page1.save', iconName: valueObj('star'), action: () => {doIt()}, display: valueObj(true)})
        await wait(10)
        expect(componentJSON(element)).toMatchSnapshot()
    }
)

test('Icon element produces output with default values where properties omitted', async () => {
        const element = createElement(Icon, {path: 'app.page1.save', iconName: 'star'})
        await wait(10)
        expect(componentJSON(element)).toMatchSnapshot()
    }
)

test('Icon does action when clicked', async () => {
    const action = jest.fn()
    let container = testContainer(createElement(Icon, {path: 'app.page1.save', iconName: 'star', action}, ))
    const iconEl = container.querySelector('button[id="app.page1.save"]')
    const user = userEvent.setup()
    await user.click(iconEl)
    expect(action).toBeCalled()
} )

