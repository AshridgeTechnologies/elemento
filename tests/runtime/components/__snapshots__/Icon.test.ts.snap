// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon element produces button output with properties supplied 1`] = `
<button
  aria-label="Star it"
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1stao2l-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Star it"
  type="button"
>
  <span
    aria-hidden={true}
    className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
  >
    star
  </span>
</button>
`;

exports[`Icon element produces empty  output with display false 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1t0xgfa-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    aria-hidden={true}
    className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
  >
    star
  </span>
</button>
`;

exports[`Icon element produces empty output with display false 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1t0xgfa-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    aria-hidden={true}
    className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
  >
    star
  </span>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Icon element produces icon output with properties supplied 1`] = `
<span
  aria-hidden={true}
  aria-label="Star it"
  className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-oo23qg-MuiIcon-root"
  id="app.page1.save"
  title="Star it"
>
  star
</span>
`;

exports[`Icon element produces output with default values where properties omitted 1`] = `
<span
  aria-hidden={true}
  className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
  id="app.page1.save"
>
  star
</span>
`;

exports[`Icon element produces output with properties supplied as state values 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-q6c508-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    aria-hidden={true}
    className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
  >
    star
  </span>
</button>
`;
